File: EntreBicis_Backend/src/main/java/cat/copernic/mbotana/entrebicis_backend/entity/GpsPoint.java
package cat.copernic.mbotana.entrebicis_backend.entity;

import java.time.LocalTime;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class GpsPoint {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private Double latitude;

    @Column
    private Double longitude;

    @Column
    private LocalTime time;

    @Column
    private Double speed;

    @Column
    private Boolean isValid;

    @ManyToOne
    @JoinColumn(name = "route_id")
    @JsonIgnoreProperties({"gpsPoints", "user"})
    private Route route;

}

File: EntreBicis_Frontend/app/src/main/java/cat/copernic/mbotana/entrebicis_frontend/class_management/gpsPoint/domain/models/GpsPoint.kt
package cat.copernic.mbotana.entrebicis_frontend.class_management.gpsPoint.domain.models

import cat.copernic.mbotana.entrebicis_frontend.class_management.route.domain.models.Route

data class GpsPoint(
    var id: Long? = null,
    var latitude: Double,
    var longitude: Double,
    var time: String?,
    var speed: Float,
    var route: Route?
)
